app/components/floating_text_input_label/index.tsx

```diff
@@ -227,18 +142,40 @@ const FloatingTextInput = forwardRef<FloatingTextInputRef, FloatingTextInputProp
         return {
             top: withTiming(toValue, {duration: 100, easing: Easing.linear}),
             fontSize: withTiming(toSize, {duration: 100, easing: Easing.linear}),
-            backgroundColor: focusedLabel || inputText ? theme.centerChannelBg : 'transparent',
-            paddingHorizontal: focusedLabel || inputText ? 4 : 0,
+            backgroundColor: showLabelOnTop ? theme.centerChannelBg : 'transparent',
+            paddingHorizontal: showLabelOnTop ? 4 : 0,
             color,
         };
     });
 
+    const focusInput = useCallback(() => {
+        inputRef.current?.focus();
+    }, []);
+
+    useImperativeHandle(ref, () => ({
+        blur: () => inputRef.current?.blur(),
+        focus: focusInput,
+        isFocused: () => inputRef.current?.isFocused() || false,
+    }), [inputRef, focusInput]);
```

Comment:
I don't get the need of this. We are just passing the functions from the reference. Why is all this needed?

Reply:
This is from the previous implementation I just maintained those parts of the code, but my guess here is the idea is to allow other components to directly trigger actions on the input such as focus, the ref can't be directly passed as the component itself uses the ref to focus the input

```tsx
<Pressable
    style={combinedTextInputContainerStyle}
    onPress={focusInput}
    disabled={!(isKeyboardInput && editable)}
    testID={`${testID}.input_container`}
>
```

Reply:
We can leave it like this for now. I will add a todo to myself to take a deeper look and see if this can be cleaned up :smile:

Reply:
this is expected